tools/clang/unittests/HLSLExec/LongVectors.cpp

```diff
@@ -195,18 +448,18 @@ void LongVector::OpTest::dispatchTestByVectorLength(
   for (auto SizeToTest : InputVectorSizes) {
     // We could create a new config for each test case with the new length, but
     // that feels wasteful. Instead, we just update the length to test.
-    TestConfig.setLengthToTest(SizeToTest);
-    testBaseMethod<DataTypeT, LongVectorOpTypeT>(TestConfig);
+    TestConfig->setLengthToTest(SizeToTest);
+    testBaseMethod<DataTypeT>(TestConfig);
   }
 }
 
-template <typename DataTypeT, typename LongVectorOpTypeT>
+template <typename DataTypeT>
 void LongVector::OpTest::testBaseMethod(
-    LongVector::TestConfig<DataTypeT, LongVectorOpTypeT> &TestConfig) {
+    std::shared_ptr<LongVector::TestConfig<DataTypeT>> &TestConfig) {
```

Comment:
`shared_ptr` is interesting here. Can you explain the overall memory model and when this'll get deallocated?

Reply:
Ah, when attempting to swap this to a member variable I recall why I didn't do that. I would need to use a std::variant to dynamically set the type of the TestConfig. And that's more cumbersome for no real gain here IMO. And for what it's worth passing it through the function parameters as a reference avoids the overhead of incrementing/decrementing the smart pointer. 

Let me know if you still have questions or concerns around that. I'll leave this comment open for now.

Reply:
Sounds good, if it makes sense, we should try a `std::unique_ptr` first since its generally safer but if not then I'm happy with this